tests/shared/src/test/scala/spire/syntax/FastForSuite.scala

```diff
@@ -60,7 +60,7 @@ class FastForSuite extends munit.FunSuite {
         b += x
       }
     }
-    assertEquals(v, 111)
+    assertEquals(v, 334) // 4 * `test`, 3 * `next`, 3 * `body`
```

Comment:
If we have to change these tests, then that means that `fastFor` semantics still don't match `cforSemantics` 😕

Reply:
I see, so it appears the cfor macro only beta reduces the function argument if it is a lambda, and not with block statements before

Reply:
I think the semantics could match if we don't implement fastFor with inline, but turn it to macro and check that the argument is a lambda.

Reply:
Would you have the bandwidth to work on that? Appreciate all your efforts so far :)

If we can make `fastFor` behave like `cfor` again, we can do away with `fastFor` altogether which would be excellent.

Reply:
(The purpose of the name change was to highlight the changed semantics.)

Reply:
I added the change I described and reverted the side-effect tests to their original state